packages/type-utils/src/isTypeReadonly.ts

```diff
@@ -198,6 +199,20 @@ function isTypeReadonlyRecurser(
     return readonlyness;
   }
 
+  if (isConditionalType(type)) {
+    const result = [type.root.node.trueType, type.root.node.falseType]
+      .map(checker.getTypeFromTypeNode)
+      .every(
+        t =>
+          seenTypes.has(t) ||
+          isTypeReadonlyRecurser(checker, t, options, seenTypes) ===
+            Readonlyness.Readonly,
+      );
+
+    const readonlyness = result ? Readonlyness.Readonly : Readonlyness.Mutable;
+    return readonlyness;
+  }
```

Comment:
Nit: you might consider writing a function with a name something like `getChildTypeParts`, since the `isUnionType` and `isConditionalType` blocks are pretty much the same _except_ for how the types are retrieved.

```ts
const childTypeParts = getChildTypeParts(type);
if (childTypeParts) {
  const result = childParts.map(checker.getTypeFromTypeNode).every(...)
  // ...
}
```

Not a blocker IMO since it's only twice 🤷, more posting for visibility if there's another case we find later on.

Reply:
I might rebase #4417 of this PR and make that change there seems it likely going to need to add 3rd case.